### PR TITLE
Add [alt] attributes to images

### LINKED ISSUE #9629: Image elements do not have [alt] attributes

### PR DESCRIPTION
**Pull request by Souvik-Cyclic**
Closes #9629 

Adds [alt] attributes to images for Improving Accessibility Score of Book Pages

### Screenshot

msedge_9tgVRGa2mc

### Stakeholders

@jimchamp

### CODE REVIEW

File: openlibrary/templates/type/edition/modal_links.html [hunk 1]
```diff
@@ -1,6 +1,6 @@
 $def with (work, edition, page_url)
 
-$def icon_link(link_class, img_src, link_text, login_redirect=True, ga_data=None):
+$def icon_link(link_class, img_src, link_text, login_redirect=True, ga_data=None, alt=""):
```
Comment (jimchamp) [inline openlibrary/templates/type/edition/modal_links.html:3]:
```suggestion
$def icon_link(link_class, img_src, link_text, login_redirect=True, ga_data=None):
```
It isn't necessary to add `alt` here, as it will be the same value for each modal link.

File: openlibrary/templates/type/edition/modal_links.html [hunk 2]
```diff
@@ -9,22 +9,22 @@
 
   $ ga_attr = 'data-ol-link-track=%s' % ga_data if ga_data else ''
   <a href="$href" class="$link_class icon-link" $ga_attr>
-    <img class="icon-link__image" src="$img_src">
+    <img class="icon-link__image" src="$img_src" alt="$alt">
```
Comment (jimchamp) [inline openlibrary/templates/type/edition/modal_links.html:12]:
```suggestion
    <img class="icon-link__image" src="$img_src" alt="">
```

File: openlibrary/templates/type/edition/modal_links.html [hunk 3]
```diff
@@ -9,22 +9,22 @@
...
     <div class="icon-link__text">$link_text</div>
   </a>
 
 <div class="modal-links">
   $if work and work.get('key', ''):
-    $ review_content = icon_link("observations-modal-link", "/static/images/icons/reviews.svg", _("Review"), ga_data="ModalLinkClick|ReviewIcon")
+    $ review_content = icon_link("observations-modal-link", "/static/images/icons/reviews.svg", _("Review"), ga_data="ModalLinkClick|ReviewIcon", alt="")
```
Comment (jimchamp) [inline openlibrary/templates/type/edition/modal_links.html:18]:
```suggestion
    $ review_content = icon_link("observations-modal-link", "/static/images/icons/reviews.svg", _("Review"), ga_data="ModalLinkClick|ReviewIcon")
```

File: openlibrary/templates/type/edition/modal_links.html [hunk 4]
```diff
@@ -9,22 +9,22 @@
...
     $:macros.ObservationsModal(work, review_content, 'sidebar-review')
 
     $:macros.StarRatings(work, edition)
 
-    $ notes_content = icon_link("notes-modal-link", "/static/images/icons/notes.svg", _("Notes"), ga_data="ModalLinkClick|NoteIcon")
+    $ notes_content = icon_link("notes-modal-link", "/static/images/icons/notes.svg", _("Notes"), ga_data="ModalLinkClick|NoteIcon", alt="")
```
Comment (jimchamp) [inline openlibrary/templates/type/edition/modal_links.html:23]:
```suggestion
    $ notes_content = icon_link("notes-modal-link", "/static/images/icons/notes.svg", _("Notes"), ga_data="ModalLinkClick|NoteIcon")
```

File: openlibrary/templates/type/edition/modal_links.html [hunk 5]
```diff
@@ -9,22 +9,22 @@
...
     $:macros.NotesModal(work, edition, notes_content, 'sidebar-notes')
 
   $if ctx.get('share_links'):
-    $ share_content = icon_link("share-modal-link", "/static/images/icons/share.svg", _("Share"), login_redirect=False, ga_data="ModalLinkClick|ShareIcon")
+    $ share_content = icon_link("share-modal-link", "/static/images/icons/share.svg", _("Share"), login_redirect=False, ga_data="ModalLinkClick|ShareIcon", alt="")
     $ share_links = ctx.get('share_links')
     $:macros.ShareModal(share_links, page_url, share_content, "share-modal-wrapper")
-</div>
+</div>
```
Comment (jimchamp) [inline openlibrary/templates/type/edition/modal_links.html:30]:
```suggestion
</div>

```
Don't remove the trailing newline character from files that you modify.

File: openlibrary/templates/type/edition/title_and_author.html [hunk 6]
```diff
@@ -52,4 +52,4 @@ <h2 class="edition-byline">
     $ share_content = icon_link("share-modal-link", "/static/images/icons/share.svg", _("Share"), login_redirect=False, ga_data="ModalLinkClick|ShareIcon")
     $ share_links = ctx.get('share_links')
     $:macros.ShareModal(share_links, page_url, share_content)
-</div>
+</div>
```
Comment (jimchamp) [inline openlibrary/templates/type/edition/title_and_author.html:55]:
```suggestion
</div>

```

File: static/css/components/nav-bar.less [hunk 7]
```diff
@@ -58,6 +58,8 @@
 
   .nav-arrow {
     padding-left: 20px;
+    background: none;
+    border: none;
```
Comment (jimchamp) [inline static/css/components/nav-bar.less:62]:
```suggestion
```
These changes seems to be unrelated.

**Review by jimchamp — CHANGES_REQUESTED** [review]
Thanks @Souvik-Cyclic!  I'm happy to merge this once the review notes have been addressed.

**Comment by Souvik-Cyclic** [comment]
@jimchamp review notes have been addressed.

File: openlibrary/templates/type/edition/modal_links.html (hunk 1 above)
Comment (Souvik-Cyclic) [reply openlibrary/templates/type/edition/modal_links.html]: removing it from here is causing some issues on the page.

Comment (jimchamp) [reply openlibrary/templates/type/edition/modal_links.html:3]: Oh!  I see why this is happening.  The `alt` keyword argument needs to be removed from the `icon_link` call on line 28.  I overlooked that one during my review...

Let me test what you have now, and I can make a final commit to this branch that will remove the argument.

File: openlibrary/macros/EditionNavBar.html [hunk 8]
```diff
@@ -32,7 +32,7 @@
     </li>
   </ul>
   $if not desktop_only:
-    <span class="nav-arrow">
-      <img src="/static/images/nav-arrow.svg" role="img">
-    </span>
+    <button class="nav-arrow" title="$_('Go to next section')">
+      <img src="/static/images/nav-arrow.svg" alt="">
+    </button>
```
Comment (jimchamp) [inline openlibrary/macros/EditionNavBar.html:37]: I didn't notice this earlier, but this drastically affected the style of the navigation arrow:
image
_`nav-arrow` now has some `<button>` styles_

Can you add `background: none;` and `border: none;` to the `nav-arrow` class, [here](https://github.com/internetarchive/openlibrary/blob/4bace241f47722c5bb8768a0dafb41a178753581/static/css/components/nav-bar.less#L59-L61)?

File: openlibrary/templates/type/edition/modal_links.html [hunk 9]
```diff
@@ -25,7 +25,7 @@
     $:macros.NotesModal(work, edition, notes_content, 'sidebar-notes')
 
   $if ctx.get('share_links'):
-    $ share_content = icon_link("share-modal-link", "/static/images/icons/share.svg", _("Share"), login_redirect=False, ga_data="ModalLinkClick|ShareIcon")
+    $ share_content = icon_link("share-modal-link", "/static/images/icons/share.svg", _("Share"), login_redirect=False, ga_data="ModalLinkClick|ShareIcon", alt="")
```
Comment (jimchamp) [inline openlibrary/templates/type/edition/modal_links.html:28]:
```suggestion
    $ share_content = icon_link("share-modal-link", "/static/images/icons/share.svg", _("Share"), login_redirect=False, ga_data="ModalLinkClick|ShareIcon")
```

**Review by jimchamp — CHANGES_REQUESTED** [review]
Thanks for the quick updates!  I noticed that the `nav-arrow` changes affected that element's styling a good bit.  Can you update the CSS for that, and push up the changes?  Apologies for missing this earlier.

__Edit:__ In order to test these changes, the `.less` files need to be compiled to CSS.  You can do this by running `docker compose exec web make css` while the webapp is running.

**Review by jimchamp — APPROVED** [review]
